README.md

```diff
@@ -139,6 +139,117 @@ Alternatively, pass `KUBECONFIG` to specify cluster access:
+
+These kind of errors mean the installation of Kubernetes elements did not complete.
+This can sometimes happen when installation of descriptors happens too quickly for changes to take effect.
+To solve the issue, identify the make target that failed and re-execute it.
+
+Once you re-execute the failing target you might see errors saying that some elements already exist, you can ignore those.
+Do verify that all pods are running before continuing with the process.
+This can be done with `kubectl get pods --all-namespaces`.
+
+Once all pods are running, look at the `Makefile` and detect which targets are still to run.
+
+E.g. this is an example where target `make install-olm` did not complete:
+
+```bash
++ kubectl create -f deploy/upstream/manifests/latest/
+unable to recognize "deploy/upstream/manifests/latest/0000_50_olm_11-olm-operators.catalogsource.yaml": no matches for kind "CatalogSource" in version "operators.coreos.com/v1alpha1"
+```
+
+E.g. this an example where the target `make install-operator` did not complete:
+
+```bash
++ kubectl apply -f https://raw.githubusercontent.com/infinispan/infinispan-operator/0.2.1/deploy/cr/cr_minimal.yaml -n local-operators
+error: unable to recognize "https://raw.githubusercontent.com/infinispan/infinispan-operator/0.2.1/deploy/cr/cr_minimal.yaml": no matches for kind "Infinispan" in version "infinispan.org/v1"
+```
+
+To be able to run the test that instantiates an Infinispan cluster with the operatorhub submission,
+the Infinispan CRD needs to be installed and the infinispan-operator needs to be running, e.g.
+
+```bash
+$ kubectl get crd
+NAME                                          CREATED AT
+infinispans.infinispan.org                    2019-04-03T09:15:48Z
+```
+
+```bash
+$ kubectl get pods --all-namespaces
+NAMESPACE         NAME                                   READY   STATUS    RESTARTS   AGE
+local-operators   infinispan-operator-5549f446f-9mqkp    1/1     Running   0          44s
+```
+
+#### Community and OpenShift 4
+
+The `operatorhub/openshift4` folder contains a `Makefile` and a series of scripts to help achieve this.
+
+**NOTE:** This section assumes you have a running OpenShift 4 cluster,
+an OpenShift client configured to talk to it,
+and you are using an administrator account.
+
+OpenShift 4 already comes with Operator Lifecycle Manager (OLM) and Operator Marketplace components installed,
+so it's not necessary to install them again.
+
+Testing on OpenShift 4 can be done automatically or manually:
+
+* In the automatic test, the operator is added to the operator hub,
+then it's installed in the defined namespace,
+and tested all in one step:
+`make all`
+
+* In the manual test, `make install-operatorsource` is called to add the operator to the operator hub.
+Installing the operator into the desired namespace is done via the Operator Hub user interface in the OpenShift console.
+Finally, the operator can be tested by calling `make test`.
+
```

Comment:
Is `make clean` to cleanup? Or I need `make delete` also?

Reply:
Clean just removes added resources to Kubernetes. Delete removes the VM altogether. I'll adjust the README.